pkg/chains/formats/intotoite6v2/taskrun/taskrun.go

```diff
@@ -0,0 +1,188 @@
+}
+
+func metadata(tro *objects.TaskRunObject) *slsa.ProvenanceMetadata {
+	m := &slsa.ProvenanceMetadata{}
+	if tro.Status.StartTime != nil {
+		utc := tro.Status.StartTime.Time.UTC()
+		m.BuildStartedOn = &utc
+	}
+	if tro.Status.CompletionTime != nil {
+		utc := tro.Status.CompletionTime.Time.UTC()
+		m.BuildFinishedOn = &utc
+	}
+	for label, value := range tro.Labels {
+		if label == attest.ChainsReproducibleAnnotation && value == "true" {
+			m.Reproducible = true
+		}
+	}
+	return m
+}
+
+// add any Git specification to materials
+func materials(tro *objects.TaskRunObject, logger *zap.SugaredLogger) []slsa.ProvenanceMaterial {
+	var mats []slsa.ProvenanceMaterial
+	gitCommit, gitURL := gitInfo(tro)
+
+	// Store git rev as Materials and Recipe.Material
+	if gitCommit != "" && gitURL != "" {
+		mats = append(mats, slsa.ProvenanceMaterial{
+			URI:    gitURL,
+			Digest: map[string]string{"sha1": gitCommit},
+		})
+		return mats
+	}
+
+	sms := artifacts.RetrieveMaterialsFromStructuredResults(tro, artifacts.ArtifactsInputsResultName, logger)
+	mats = append(mats, sms...)
+
+	if tro.Spec.Resources == nil {
+		return mats
+	}
+
+	// check for a Git PipelineResource
+	for _, input := range tro.Spec.Resources.Inputs {
+		if input.ResourceSpec == nil || input.ResourceSpec.Type != v1alpha1.PipelineResourceTypeGit {
+			continue
+		}
+
+		m := slsa.ProvenanceMaterial{
+			Digest: slsa.DigestSet{},
+		}
+
+		for _, rr := range tro.Status.ResourcesResult {
+			if rr.ResourceName != input.Name {
+				continue
+			}
+			if rr.Key == "url" {
+				m.URI = attest.SPDXGit(rr.Value, "")
+			} else if rr.Key == "commit" {
+				m.Digest["sha1"] = rr.Value
+			}
+		}
+
+		var url string
+		var revision string
+		for _, param := range input.ResourceSpec.Params {
+			if param.Name == "url" {
+				url = param.Value
+			}
+			if param.Name == "revision" {
+				revision = param.Value
+			}
+		}
+		m.URI = attest.SPDXGit(url, revision)
+		mats = append(mats, m)
+	}
+	return mats
+}
+
+// gitInfo scans over the input parameters and looks for parameters
+// with specified names.
+func gitInfo(tro *objects.TaskRunObject) (commit string, url string) {
+	// Scan for git params to use for materials
+	if tro.Status.TaskSpec != nil {
+		for _, p := range tro.Status.TaskSpec.Params {
+			if p.Default == nil {
+				continue
+			}
+			if p.Name == attest.CommitParam {
+				commit = p.Default.StringVal
+				continue
+			}
+			if p.Name == attest.URLParam {
+				url = p.Default.StringVal
+			}
+		}
+	}
+
+	for _, p := range tro.Spec.Params {
+		if p.Name == attest.CommitParam {
+			commit = p.Value.StringVal
+			continue
+		}
+		if p.Name == attest.URLParam {
+			url = p.Value.StringVal
+		}
+	}
+
+	for _, r := range tro.Status.TaskRunResults {
+		if r.Name == attest.CommitParam {
+			commit = r.Value.StringVal
+		}
+		if r.Name == attest.URLParam {
+			url = r.Value.StringVal
+		}
+	}
+
+	url = attest.SPDXGit(url, "")
+	return
+}
```

Comment:
Can any of this be shared with the existing intoto implementation?

Reply:
I think materials can be shared.